estimagic/examples/numdiff_example_functions_np.py

```diff
@@ -0,0 +1,77 @@
+"""Functions with known gradients, jacobians or hessians.
+
+All functions take a numpy array with parameters as their first argument.
+
+Example inputs for the binary choice functions are in binary_choice_inputs.pickle.
+They come from the statsmodels documentation: https://tinyurl.com/y4x67vwl
+We pickled them so we don't need statsmodels as a dependency.
+
+"""
+import numpy as np
+from scipy.stats import norm
+
+FLOAT_EPS = np.finfo(float).eps
+
+# ======================================================================================
+# Logit
+# ======================================================================================
+
+
+def logit_loglike(params, y, x):
+    return logit_loglikeobs(params, y, x).sum()
+
+
+def logit_loglikeobs(params, y, x):
+    q = 2 * y - 1
+    return np.log(1 / (1 + np.exp(-(q * np.dot(x, params)))))
+
+
+def logit_loglike_gradient(params, y, x):
+    c = 1 / (1 + np.exp(-(np.dot(x, params))))
+    return np.dot(y - c, x)
+
+
+def logit_loglikeobs_jacobian(params, y, x):
+    c = 1 / (1 + np.exp(-(np.dot(x, params))))
+    return (y - c)[:, None] * x
+
+
+def logit_loglike_hessian(params, y, x):
+    c = 1 / (1 + np.exp(-(np.dot(x, params))))
+    return -np.dot(c * (1 - c) * x.T, x)
```

Comment:
These functions look identical to those in examples/logit.py.

Reply:
Yes, but they have numpy instead of pandas params. In the long run we will replace them with functions that are harder to differentiate numerically, so it makes sense to have separate example functions for optimization and differentiation.